appserver/extras/payara-micro/payara-micro-boot/src/main/java/fish/payara/micro/boot/PayaraMicroBoot.java

```diff
@@ -86,6 +87,25 @@
      * @return
      */
     PayaraMicroBoot addRepoUrl(String... URLs);
+    
+    /**
+     * Adds a command to run before booting Payaa Micro.
+     * Not that only a limited set of commands will run here i.e. {@code set}.
+     * @param command Command to add
+     */
+    public void addPreBootCommand(BootCommand command);
```

Comment:
Did you consider callbacks or other means of passing command result back to the caller, as was requested in the issue?